ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivity.kt

```diff
@@ -49,6 +60,28 @@ internal class PaywallActivity : ComponentActivity(), PaywallListener {
         }
     }
 
+    @Parcelize
+    private data class SdkConfigArgs(
+        val apiKey: String,
+        val appUserId: String?,
+        val purchasesAreCompletedBy: PurchasesAreCompletedBy,
+        val showInAppMessagesAutomatically: Boolean,
+        val store: Store,
+        val diagnosticsEnabled: Boolean,
+        val verificationMode: EntitlementVerificationMode,
+        val dangerousSettings: DangerousSettings,
+        val pendingTransactionsForPrepaidPlansEnabled: Boolean,
+    ) : Parcelable
+
+    private fun getSdkConfigArgs(savedInstanceState: Bundle): SdkConfigArgs? {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            savedInstanceState.getParcelable(SDK_CONFIG_EXTRA, SdkConfigArgs::class.java)
```

Comment:
I always found the warning in the docs for this one a bit scary: 

> **Warning:** the class that implements `Parcelable` has to be the immediately enclosing class of the runtime type of its CREATOR field (that is, `Class.getEnclosingClass()` has to return the parcelable implementing class), otherwise this method might throw an exception. If the `Parcelable` class does not enclose the CREATOR, use the deprecated `getParcelable(String)` instead.

We should be good, but we can also just use the deprecated variant to avoid it altogether. If we ever change things around in the future, we might run into this exception.